torch/fx/_pytree.py

```diff
@@ -36,30 +36,23 @@ def _deregister_pytree_flatten_spec(
 def tree_flatten_spec(
     pytree: PyTree,
     spec: TreeSpec,
-    exact_structural_match: bool = False,
 ) -> list[Any]:
     if spec.is_leaf():
         return [pytree]
-    if spec.type not in SUPPORTED_NODES:
-        raise RuntimeError(
-            f"{type(pytree)} does not have a flatten_fn_spec associated with it. Please register one with "
-            "torch.fx._pytree.register_pytree_flatten_spec.  If you have serialized your model, make "
-            "sure that any custom pytrees have been registered before loading it.",
-        )
-    flatten_fn_spec = SUPPORTED_NODES[spec.type]
-    child_pytrees = flatten_fn_spec(pytree, spec)
-    if exact_structural_match:
-        flatten_fn_exact_match_spec = SUPPORTED_NODES_EXACT_MATCH[spec.type]
-        if flatten_fn_exact_match_spec and not flatten_fn_exact_match_spec(
-            pytree,
-            spec,
-        ):
-            raise RuntimeError(f"Cannot flatten pytree {pytree}, given spec: {spec}")
-    result = []
-    for child, child_spec in zip(child_pytrees, spec.children_specs):
-        flat = tree_flatten_spec(child, child_spec, exact_structural_match)
-        result += flat
-    return result
+    # I guess these exist for BC, FC reasons.
+    # In general, we should be able to directly
+    # use pytree tree flattener to flatten them,
+    # as export serializes the pytree seperately.
+    # Will remove it in follow up PR.
+    if spec.type in SUPPORTED_NODES:
+        flatten_fn_spec = SUPPORTED_NODES[spec.type]
+        child_pytrees = flatten_fn_spec(pytree, spec)
+        result = []
+        for child, child_spec in zip(child_pytrees, spec.children_specs):
+            flat = tree_flatten_spec(child, child_spec)
+            result += flat
+        return result
+    return list(tree_flatten(pytree)[0])
```

Comment:
Can you add a check that the spec after flattening is the same as the spec being passed in?

If it is the same, then good. If it is different, we should raise an error.

Reply:
Updated!